src/Spawner.cpp

```diff
@@ -132,18 +139,21 @@ Spawner::ProcessHandle::ProcessHandle(Deployment *deployment, bool redirectOutpu
         {
             throw std::runtime_error("Error, log directory '" + logDir + "' does not exist, but it should !");
         }
-        processName = deployment->getName();
 
-        if (!textLogFileName.empty())
-        {
-            redirectOutput(logDir + "/" + textLogFileName + ".txt");
-        }
-        else
-        {
-            redirectOutput(logDir + "/" + processName + "-" + boost::lexical_cast<std::string>(getpid()) + ".txt");
-        }
+        redirectOutput(logDir + "/" + textLogFileName + ".txt");
     }
     
+    //set ORO_LOGFILE so the new deployment logs to its own orocos.log file
+    if(setenv("ORO_LOGFILE", (logDir + "/" + textLogFileName + "-orocos.log").c_str(), true) != 0)
+    {
+        LOG_WARN_S << "Could not set ORO_LOGFILE environment variable";
+    }
+
+    //close all filehandles above 2 (stderr)
+    for(unsigned int i = 3; i < 1024; i++) {
```

Comment:
Could you elaborate on this and extend code documentation accordingly.

Which file descriptors are expected to be open and need to be closed?

Is the intention to close the file descriptor of the text log file created here https://github.com/rock-cpp/orocos_cpp/blob/c7a6a4fb4a0c2285383a3ad65fa49f6991a3b966/src/Spawner.cpp#L427 (should not be necessary because of dup2, I think)

Is it for the ORO_LOG_FILE, that is not closed correctly by RTT? Why?

Some other reason?

If a clean closingis not possible for some reason and we really have to blindly close all files:
Is there no more elegant way for closing that file without erno being set due to unsuccessful close? (e.g. https://stackoverflow.com/a/26672559)